### PR TITLE
Remove some unnecessary List-related allocation from EventSource

### LINKED ISSUE #43989: Reduce List<object> allocations by passing the initial array
Not sure if this is the right repository, or if still relevant as I could not find the source code of EventSource here.

Some minor improvement I noticed while benchmarking allocations on our code, I think could be traced to [the call to Add on the list in this line](https://github.com/microsoft/referencesource/blob/08b84d13e81cfdbd769a557b368539aac6a9cb30/mscorlib/system/diagnostics/eventing/eventsource.cs#L1423)

image

`````csharp
    List<object> arg = new List<object>();
    arg.Add(msg);
`````

could be changed to:

`````csharp
    List<object> arg = new List<object>(new []{ msg });
`````

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/43989

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs [hunk 1]
```diff
@@ -5764,21 +5760,6 @@ private void WriteMessageAttrib(StringBuilder stringBuilder, string elementName,
             return value;
         }
 
-        /// <summary>
-        /// There's no API to enumerate all languages an assembly is localized into, so instead
-        /// we enumerate through all the "known" cultures and attempt to load a corresponding satellite
-        /// assembly
-        /// </summary>
-        /// <returns></returns>
-        private static List<CultureInfo> GetSupportedCultures()
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:5773]: This is interesting, I am not sure we had such code in the first place which just create a list and add the UI culture. 

I guess there were fallback mechanism before which may allow adding more cultures to the list and then get removed? 

anyway, this change looks good to me.

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
CC @noahfalk

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:5773]: > I guess there were fallback mechanism before which may allow adding more cultures to the list and then get removed?

Yeah, it used to contain this:
https://github.com/dotnet/coreclr/blob/144c15ead4c88a8c3e8840cb8b8b0a87e03019f8/src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs#L6645-L6651

**Comment by GrabYourPitchforks** [comment]
@stephentoub Did you intend for this PR to resolve https://github.com/dotnet/runtime/issues/43989? That issue seemingly tracks a new feature request.

**Comment by stephentoub** [comment]
> Did you intend for this PR to resolve #43989? That issue seemingly tracks a new feature request.

I don't see any new feature request in the comments from the dev who opened the issue?
https://github.com/dotnet/runtime/issues/43989#issue-731829542

What feature?

**Comment by GrabYourPitchforks** [comment]
Ah, I misinterpreted that issue as a "please allow me to provide my own backing array to `List<T>`." Carry on! :)

**Comment by stephentoub** [comment]
:smile:

**Pull request merged**